modules/apps/frontend-js/frontend-js-walkthrough-web/src/main/resources/META-INF/resources/Walkthrough.js

```diff
@@ -203,52 +228,63 @@ const Step = ({
 		setCurrentAlignment(defaultPositioning);
 	}, [defaultPositioning]);
 
-	const align = useCallback(() => {
-		if (popoverVisible && popoverRef.current && memoizedTrigger) {
-			const points = ALIGNMENTS_MAP[currentAlignment];
-
-			const alignment = doAlign({
-				offset: [OVERLAY_OFFSET_X, OVERLAY_OFFSET_Y],
-				overflow: {
-					adjustX: true,
-					adjustY: true,
-				},
-				points,
-				sourceElement: popoverRef.current,
-				targetElement: memoizedTrigger,
-			});
-
-			const alignmentString = alignment.points.join('');
-
-			const pointsString = points.join('');
-
-			if (alignment.overflow.adjustX) {
-				setCurrentAlignment(ALIGNMENTS_GUESS_MAP[alignmentString]);
+	const align = useCallback(
+		(boundingRect) => {
+			if (popoverVisible && popoverRef.current && memoizedTrigger) {
+				const points = ALIGNMENTS_MAP[currentAlignment];
+
+				const alignment = doAlign({
+					offset: [OVERLAY_OFFSET_X, OVERLAY_OFFSET_Y],
+					overflow: {
+						adjustX: true,
+						adjustY: true,
+					},
+					points,
+					sourceElement: popoverRef.current,
+					targetElement: memoizedTrigger,
+				});
+
+				const alignmentString = alignment.points.join('');
+
+				const pointsString = points.join('');
+
+				if (alignment.overflow.adjustX) {
+					setCurrentAlignment(ALIGNMENTS_GUESS_MAP[alignmentString]);
+				}
+				else if (pointsString !== alignmentString) {
+					setCurrentAlignment(
+						ALIGNMENTS_INVERSE_MAP[alignmentString]
+					);
+				}
+
+				if (
+					!darkbg &&
+					previousTrigger &&
+					previousTrigger !== memoizedTrigger
+				) {
+					memoizedTrigger.classList.add(
+						'lfr-walkthrough-element-shadow'
+					);
+
+					previousTrigger.classList.remove(
+						'lfr-walkthrough-element-shadow'
+					);
+				}
+
+				if (!isVisibleInViewport(popoverRef?.current, boundingRect)) {
+					popoverRef?.current?.scrollIntoView();
```

Comment:
Here would be just a modification in this use case that it would not be necessary to use Optional chaining since you are already checking if there is a value in `popoverRef.current`, also a `useRef` will always have set the `.current`.

Reply:
You're right, changing.